osmomath/decimal.go

```diff
@@ -332,6 +343,20 @@ func (d BigDec) Quo(d2 BigDec) BigDec {
 	return BigDec{chopped}
 }
 
+// mutative quotient
+func (d BigDec) QuoMut(d2 BigDec) BigDec {
+	// multiply precision twice
+	mul := new(big.Int).Mul(d.i, precisionReuse)
```

Comment:
We should not allocate new int. One of the goals of the mutative versions is to gain speed up by avoiding reallocations and copying.

Please see:
https://github.com/osmosis-labs/cosmos-sdk/blob/48fe175f983925ff6202d492a363d2b2e3a62497/types/decimal.go#L343-L352